src/trackCopiedText.js

```diff
@@ -1,18 +1,57 @@
-/**
- * Tracking text copied by the user.
- * 
- * Adds event listener that monitors all copy events and sends text copied at the time as custom event.
- */
-
- export default (subscribe) => {
-    
-    const processCopyEvent = () => {
-        let copiedItemText = getSelectedText();
-        var eventData = ["trackEvent","User interaction","Copying text",copiedItemText];
-        subscribe(eventData);   
-    };
-    
-    const getSelectedText = () => window.getSelection()?.toString() ?? '';
-    
-    window.addEventListener("copy", processCopyEvent);
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
+/**
+ * Tracking text copied by the user.
+ *
+ * Adds event listener that monitors all copy events and sends text copied at the time as custom event.
+ */
+
+export default (subscribe, {filteredElements}) => {
+
+    const processCopyEvent = () => {
+        let copiedItemText = getSelectedText();
+        if(copiedItemText){
+            var eventData = ["trackEvent","User interaction","Copying text",copiedItemText];
+            subscribe(eventData);
+        }
+    };
+
+    const getSelectedText = () => {
+       let filter = false;
+       //check text node parent elements
+       if(window.getSelection()?.baseNode instanceof Text) {
+        filteredElements.forEach((filteredElement) => {
+            let selectedElementNodeName = window.getSelection()?.baseNode?.nodeName;
+            let selectedElementParentElement = window.getSelection()?.baseNode?.parentElement;
+            if(selectedElementNodeName === "#text") {
+                if(selectedElementParentElement.nodeName.toUpperCase() === filteredElement.toUpperCase()) {
+                    filter = true;
+                    return;
+                }
+            }
+        });
+       }
+       //check input/textarea elements
+       if(window.getSelection()?.baseNode instanceof HTMLElement) 
+       {
+        filteredElements.forEach((filteredElement) => {
+            let filteredElementArray = window.getSelection()?.baseNode?.querySelectorAll(filteredElement);
+            if(filteredElementArray?.length) {
+                filteredElementArray.forEach((item) => {
+                    if(item?.value.includes(window.getSelection()?.toString())) 
+                    {
+                        filter = true;
+                        return;

```

Comment:
We don't need that filter flag to be set up to return it in the end. We can simple return that `''` value in here

Reply:
Its in a forEach loop so returning the value here won't work, I've removed the return however since it doesn't stop the loop anyway.